src/routes/Measurements/components/nettests/Dash.js

```diff
@@ -0,0 +1,153 @@
+import React from 'react'
+import { FormattedMessage } from 'react-intl'
+
+import { mlabServerToCountry, mlabServerToName } from '../../../../util/nettest'
+
+/*
+ * This table is derived from:
+ * https://support.google.com/youtube/answer/1722171?hl=en-GB
+ */
+
+const minimumBitrateForVideo = [
+  {
+    'sfr_min_bitrate': 600,
+    'hfr_min_bitrate': 1000,
+    'type': '240p'
+  },
+  {
+    'sfr_min_bitrate': 1000,
+    'hfr_min_bitrate': 1500,
+    'type': '360p'
+  },
+  {
+    'sfr_min_bitrate': 2500,
+    'hfr_min_bitrate': 4000,
+    'type': '480p'
+  },
+  {
+    'sfr_min_bitrate': 5000,
+    'hfr_min_bitrate': 7500,
+    'type': '720p (HD)'
+  },
+  {
+    'sfr_min_bitrate': 8000,
+    'hfr_min_bitrate': 12000,
+    'type': '1080p (full HD)'
+  },
+  {
+    'sfr_min_bitrate': 16000,
+    'hfr_min_bitrate': 24000,
+    'type': '1440p (2k)'
+  },
+  {
+    'sfr_min_bitrate': 35000,
+    'hfr_min_bitrate': 53000,
+    'type': '2160p (4k)'
+  },
+]
+
+const getOptimalVideoRate = (test_keys) => {
+  let optimalRate = minimumBitrateForVideo[0]
+
+  minimumBitrateForVideo.forEach((rate) => {
+    if (test_keys.simple.median_bitrate >= rate['sfr_min_bitrate']) {
+      optimalRate = rate
+    }
```

Comment:
So, I have written this stripped down file:

```js
$ cat example.js 

/*
 * This table is derived from:
 * https://support.google.com/youtube/answer/1722171?hl=en-GB
 */
const minimumBitrateForVideo = [
  {
    'sfr_min_bitrate': 600,
    'hfr_min_bitrate': 1000,
    'type': '240p'
  },
  {
    'sfr_min_bitrate': 900,
    'hfr_min_bitrate': 1500,
    'type': '360p'
  },
]

const getOptimalVideoRate = (test_keys) => {
  let optimalRate = minimumBitrateForVideo[0]

  minimumBitrateForVideo.forEach((rate) => {
    if (test_keys.simple.median_bitrate >= rate['sfr_min_bitrate']) {
      optimalRate = rate
    }
  })
  return optimalRate
}

const getOptimalVideoRateEx = (test_keys) => {
  let optimalRate = null;
  minimumBitrateForVideo.forEach((rate) => {
    // Make sure we select the lowest speed bucket in case they overlap
    if (optimalRate === null &&
        test_keys.simple.median_bitrate >= rate['sfr_min_bitrate']) {
      optimalRate = rate
    }
  })
  if (optimalRate === null) {
    optimalRate = minimumBitrateForVideo[0]
  }
   return optimalRate
}

const input = {
  "simple": {
    "median_bitrate": 911,
  },
}

console.log(getOptimalVideoRate(input).type)
console.log(getOptimalVideoRateEx(input).type)
```

When running it, I get:

```
$ node example.js 
360p
240p
```

My initial comment was like: I *assumed* the function wanted to compute the *lowest* bucked matching the speed. So, I commented that in case of overlapping qualities &mdash; which does not happen in the real file but happened in this example &mdash; the implementation does not select the lowest but the highest bucket. And, hence, I changed the implementation.

It's true that the current input file is such that there is no overlapping (so nothing changes). But, I believe we should decide what to report to the user. I am not super happy about reporting the highest bucket, because, in the event where there is great overlapping, we would end up reporting something non realistic. (I recall that elsewhere I saw overlapping ranges.)